### ISSUE TITLE
Delay not working

### ISSUE THREAD

**Issue by BrosseauSylvain** [edoardottt/cariddi#163]
**Problem**
The -d option for delay is not working properly

```
$ echo https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/ | cariddi -d 2         
                 _     _     _ _ 
                (_)   | |   | (_)
   ___ __ _ _ __ _  __| | __| |_ 
  / __/ _` | '__| |/ _` |/ _` | |
 | (_| (_| | |  | | (_| | (_| | |
  \___\__,_|_|  |_|\__,_|\__,_|_| v1.3.5

 > github.com/edoardottt/cariddi
 > edoardoottavianelli.it
========================================
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/post?postId=2
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/post?postId=1
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/post?postId=3
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/post?postId=4
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/post?postId=5
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/resources/labheader/css/academyLabHeader.css
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/resources/labheader/js/labHeader.js
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/resources/images/blog.svg
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/image/blog/posts/26.jpg
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/resources/css/labsBlog.css
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/image/blog/posts/18.jpg
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/image/blog/posts/9.jpg
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/image/blog/posts/15.jpg
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/image/blog/posts/48.jpg
https://0a2500f1033d0c0e83370f0d005500fa.web-security-academy.net/resources/images/avatarDefault.svg
```
![image](https://github.com/user-attachments/assets/a9109001-b7d9-4ce7-8de0-5cdd0fc27491)
**To Reproduce**
```
echo $URL | cariddi -d 1000000000
```

**Expected behavior**

-d should ensure that a minimum delay is present between 2 requests as it prevent being blocked by a firewall rate limit.

**Comment by edoardottt** [comment]
Hi! 
The delay is working fine, but cariddi is a multithreaded tool. You're setting a delay for each thread, but the threads are 20 by default. Try setting the concurrency to 1 (`-c 1`).
See also https://github.com/edoardottt/cariddi/issues/155

**Comment by BrosseauSylvain** [comment]
I've tested it on my localhost and it works just fine with ```-c 1```.
But on my port swigger lab it doesn't work even with the ```-c 1``` option.

**Comment by edoardottt** [comment]
Probably all the links you crawl are present on the main page. Recall that the delay option is not a rate limit, as explained also in the issue I linked. 
cariddi uses [colly](https://github.com/gocolly/colly) under the hood and it doesn't have a rate limit option. If you want to contribute, just open a PR : )

**Comment by BrosseauSylvain** [comment]
Oh ok thank you for your answer ! I will maybe do it if I continue using cariddi as I've just discovered your tool.

**Comment by edoardottt** [comment]
Thanks @BrosseauSylvain, I'll close this issue for now. 
For anything else just open another issue :)
